src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -787,23 +816,54 @@ void Graph::AllocateWithReuse() {
     }
 
     if (!undefinedBoxes.empty()) {
+        if (!syncNodesInds.empty()) {
+            //We have to extend the lifespan of thensors that are crossing a sync point border in order to save
+            //the intermediate computation results from possible loss due to the tensor resize
+            std::vector<int> vecIntervals = {0};
+            for (const auto& item : syncNodesInds) {
+                vecIntervals.push_back(item.first->execIndex);
+            }
+            std::sort(vecIntervals.begin(), vecIntervals.end());
+            for (auto& box : undefinedBoxes) {
+                if (-1 == box.finish) {
+                    continue;
+                }
+                auto itr_upper = std::upper_bound(vecIntervals.begin(), vecIntervals.end(), box.finish, [](int y, int x) { return y <= x;});
+                auto itr_lower = std::lower_bound(vecIntervals.begin(), vecIntervals.end(), box.start);
+                if (itr_lower != itr_upper) { // across sections
+                    if (itr_upper == vecIntervals.end()) {
+                        box.finish = -1;
+                    } else {
+                        box.finish = *itr_upper;
+                    }
+                }
+            }
+        }
```

Comment:
Here again, why do we need this lifespan change? Any dedicate example of failure if without it?

Reply:
In this PR we changed dynamic shapes processing completely. We separated shape inference and parameters preparation stages to a different pass, which happens before the actual computations. Since we have sync points, the graph is actually split by parts, for which we initially perform the dyn shapes preparation pass, then inference pass, then switch to the next part of the graph, one by one. With this approach we have situations when we have to preserve intermediate computation results. To this end, we have to recalculate the lifespan of boxes that cross the sync point boundaries so that we do not lost the intermediate calculations results.

Reply:
Even with the new approach, the execute order of nodes are still the same as before. Then how does the violation happen and intermediate computations results gets lost?

Reply:
Due to memory reuse. When some tensor from let's say second subraph shares the mem manager with some tensor from the first subgraph, and these data are used in the second subgraph, tensor resize in the second subgraph may trigger memory reallocation in the memory manager, which invalidate the previous memory blob where the intermediate data are located.

Reply:
The problem is not in overlapping. The problem is in data lost due to the memory resize. Since we do shape inference on the whole subgraph, and run the actual operations only after that. During this shape inference the underlying memory shared with a tensor from the first subgraph may be resized, and data, needed for calculation are lost, before they are read by the second subgraph nodes, exactly because the shape infer for the whole subgraph happens before the execution.